### ISSUE TITLE
Copy/paste is not working in wlroots-based window managers

### ISSUE THREAD

**Issue by tomKPZ** [vmware/open-vm-tools#510]
Using VMware Workstation 16 pro 16.1.1 build-17801498 with an Arch Linux guest OS running the Sway window manager.  The issue is that wlroots (the project sway is based on) blocks clipboard requests when the xwayland client doesn't have focus [1] due to security concerns.  So the requests from [2] are getting blocked.  The solution recommended [3] by a wlroots member is to use the wlr-data-control-unstable-v1 protocol to interact with the clipboard, like how wl-copy/wl-paste [4] does.

[1] https://github.com/swaywm/wlroots/commit/b884025558e750268a06818dc63bc46716c75843
[2] https://github.com/vmware/open-vm-tools/blob/master/open-vm-tools/services/plugins/dndcp/copyPasteUIX11.cpp
[3] https://github.com/swaywm/wlroots/pull/2887
[4] https://github.com/bugaevc/wl-clipboard

**Comment by PaTHml** [comment]
Yes, this is a known issue, Wayland support is under development.
Status is unchanged from Issue #443

Need to double check if that status includes wlroots.
Thank you for the suggestion and links.

Just in case:
The workaround from KB: https://kb.vmware.com/s/article/74671 is to Log in with X11 session.

**Comment by tomKPZ** [comment]
> Are you having any issues with a (menu) copy/paste of either files or text?

Menu paste of text is OK (it appears to emulate keypresses).  Though menu text copy and menu file copy/paste is not.

**Comment by ddnomad** [comment]
Any update on this one? Were trying to switch my development VM to Wayland but sadly this issue is a showstopper.

Is there anything that can be done to assist the development?

**Comment by raknerak** [comment]
We're still waiting for wayland support for years. I prefer wayland instead of x11 however, this issue is blocking when I use windows as a guest occasionally over ubuntu 22.04. vmware version is the latest: 17.5.2 build-23775571. I'm looking forward to get solution on this issue.

**Comment by krisztianfekete** [comment]
Hello, I hit this problem, and have a potential solution for thrill-seekers: https://github.com/krisztianfekete/clipway/, use it at your own risk!

**Comment by aSapien** [comment]
@krisztianfekete it worked flawlessly on my NixOS VMWare machine. Thank you!
